src/components/contestants.jsx

```diff
@@ -0,0 +1,245 @@
+
+import "./contestants.less";
+import "./jury.less";
+
+
+export default React.createClass({
+  displayName: "Participanti",
+
+  getInitialState: function() {
+    let isMobile = Math.max(document.documentElement.clientWidth, window.innerWidth || 0) <= 768;
+
+    return {
+     projects: [
+       {
+        "title": "Time Travel",
+        "authors": ["claudia rujoiu", "dan stoian"],
+        "county": "hunedoara",
+        "comments": 7,
+        "category": "utility",
+        "slug": "utilitar"
+       },
+       {
+        "title": "Acționarea automată și securizată a Căilor de Access",
+        "authors": ["adriana-patricia blaj", "robert preda"],
+        "county": "arad",
+        "comments": 5,
+        "category": "robots",
+        "slug": "roboți"
+       },
+       {
+        "title": "Romania e Okay!",
+        "authors": ["diana elena ghinea"],
+        "county": "bucurești",
+        "comments": 1,
+        "category": "educational",
+        "slug": "educațional"
+       },
+       {
+        "title": "Mica Unire",
+        "authors": ["georgiana raluca bucutea", "roxana patricia datcu"],
+        "county": "brașov",
+        "comments": 13,
+        "category": "media",
+        "slug": "multimedia"
+       },
+       {
+        "title": "Velocity Reporter",
+        "authors": ["mihaela costache", "bogdan nechifor"],
+        "county": "brăila",
+        "comments": 5,
+        "category": "web",
+        "slug": "web"
+       },
+       {
+        "title": "Acționarea automată și securizată a Căilor de Access",
+        "authors": ["adriana-patricia blaj", "robert preda"],
+        "county": "arad",
+        "comments": 5,
+        "category": "robots",
+        "slug": "roboți"
+       },
+       {
+        "title": "Romania e Okay!",
+        "authors": ["diana elena ghinea"],
+        "county": "bucurești",
+        "comments": 1,
+        "category": "educational",
+        "slug": "educațional"
+       },
+       {
+        "title": "Mica Unire",
+        "authors": ["georgiana raluca bucutea", "roxana patricia datcu"],
+        "county": "brașov",
+        "comments": 13,
+        "category": "media",
+        "slug": "multimedia"
+       },
+       {
+        "title": "Velocity Reporter",
+        "authors": ["mihaela costache", "bogdan nechifor"],
+        "county": "brăila",
+        "comments": 5,
+        "category": "web",
+        "slug": "web"
+       }],
+     showGrid: isMobile ? false : true,
+     showTable: isMobile ? true : false,
+     showCategory: {
+        "educational": true,
+        "utility": true,
+        "web": true,
+        "media": true,
+        "robots": true
+     }
+   };
+
+  },
+
+  showGrid() {
+    this.setState({
+      showTable: false,
+      showGrid: true
+    });
+  },
+
+  showTable() {
+    this.setState({
+      showTable: true,
+      showGrid: false
+    });
+  },
+
+  toggleCategory(category) {
+    var newStates = this.state.showCategory;
```

Comment:
Wrong. În Javascript în momentul ăsta newStates este o referință la `this.state.showCategory`, iar tu pe linia de jos mutezi `this.state` ceea ce e un anti-pattern în React.

Sunt două lucruri pe care le poți face:

* `_.clone` din Lodash
* [immutability helpers](https://facebook.github.io/react/docs/update.html)